contracts/schemes/ContributionRewardExt.sol

```diff
@@ -0,0 +1,452 @@
+pragma solidity 0.5.13;
+
+import "@daostack/infra/contracts/votingMachines/IntVoteInterface.sol";
+import "@daostack/infra/contracts/votingMachines/VotingMachineCallbacksInterface.sol";
+import "../votingMachines/VotingMachineCallbacks.sol";
+import "../libs/SafeERC20.sol";
+
+
+/**
+ * @title A scheme for proposing and rewarding contributions to an organization
+ * @dev An agent can ask an organization to recognize a contribution and reward
```

Comment:
So (after reading the code) this is an extended version of the ContriutionReward contract, with the possibility to assign a rewarder, which, after the contributionreward has been accepted, can then later distribute the assets as it would like.

So perhaps call it "ContributionRewardWithRewarder"?

To me, it seems a bit crazy do stuff all of this logic in a single contract (it is an enormous, complex, contract) 

If I understand it correctly, if the rewards were all transferable tokens (i.e. no rep), we could have a much cleaner setup by using our original contribution reward, and transfer the assets to a beneficiary contract  that manages the payouts for the competition.  But (I think) this will not be possible bc the rights to mint rep are difficult to manage.

(which makes me think - not for this PR, but for arc2.0 - instead of doing all this administration of keeping track fo which schemes have the right to mint which rep, it could perhaps be easier and more secure to work with a "rightToMintRep"  token)

Reply:
> So (after reading the code) this is an extended version of the ContriutionReward contract, with the possibility to assign a rewarder, which, after the contributionreward has been accepted, can then later distribute the assets as it would like.
> 
> So perhaps call it "ContributionRewardWithRewarder"?
> 
> To me, it seems a bit crazy do stuff all of this logic in a single contract (it is an enormous, complex, contract)
> 
> If I understand it correctly, if the rewards were all transferable tokens (i.e. no rep), we could have a much cleaner setup by using our original contribution reward, and transfer the assets to a beneficiary contract that manages the payouts for the competition. But (I think) this will not be possible bc the rights to mint rep are difficult to manage.
> 
> (which makes me think - not for this PR, but for arc2.0 - instead of doing all this administration of keeping track fo which schemes have the right to mint which rep, it could perhaps be easier and more secure to work with a "rightToMintRep" token)

yes. If a rep was not a req here we could teoreticlyl use a normal cr with the beneficiary the competition scheme / or other